src/appshell/view/appmenumodel.cpp

```diff
@@ -42,23 +42,18 @@ void AppMenuModel::load()
             << toolsItem()
             << helpItem();
 
-    globalContext()->currentMasterNotationChanged().onNotify(this, [this]() {
-        load();
-
-        currentMasterNotation()->notation()->notationChanged().onNotify(this, [this]() {
-            load();
-        });
-
-        currentMasterNotation()->notation()->interaction()->selectionChanged().onNotify(this, [this]() {
-            load();
-        });
-    });
+    setupConnections();
 
     emit itemsChanged();
 }
 
-void AppMenuModel::handleAction(const QString& actionCode)
+void AppMenuModel::handleAction(const QString& actionCode, const QVariant& actionData)
 {
+    if (actionCode == "file-open") {
```

Comment:
I don't like the idea to write a logic that depends on concrete codes here. This approach has poor scalability (you should add else-if for all actions that have arguments). Better write an abstract code for handling actions:

args = action.args

if (action.args.isEmpty) {
dispatch(actionCode)
} else {
dispatch(actionCode, action.args)
}

Reply:
or just `dispatch(actionCode, action.args)`

Reply:
args is QVariant, not ActionData
need to restore the types of arguments, and this is possible only by code

Reply:
thanks 
I am not passing arguments to qml now. The index of the action in the group comes from qml, by which I find the action locally